src/ios/PushPlugin.m

```diff
@@ -353,35 +359,43 @@ - (UNNotificationAction *)createAction:(NSDictionary *)dictionary {
 
 - (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
     if (self.callbackId == nil) {
-        NSLog(@"Unexpected call to didRegisterForRemoteNotificationsWithDeviceToken, ignoring: %@", deviceToken);
+        NSLog(@"[PushPlugin] Unexpected call to didRegisterForRemoteNotificationsWithDeviceToken");
         return;
     }
-    NSLog(@"Push Plugin register success: %@", deviceToken);
 
-#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
-    // [deviceToken description] is like "{length = 32, bytes = 0xd3d997af 967d1f43 b405374a 13394d2f ... 28f10282 14af515f }"
-    NSString *token = [self hexadecimalStringFromData:deviceToken];
-#else
-    // [deviceToken description] is like "<124686a5 556a72ca d808f572 00c323b9 3eff9285 92445590 3225757d b83967be>"
-    NSString *token = [[[[deviceToken description] stringByReplacingOccurrencesOfString:@"<"withString:@""]
-                        stringByReplacingOccurrencesOfString:@">" withString:@""]
-                       stringByReplacingOccurrencesOfString: @" " withString: @""];
-#endif
+    NSLog(@"[PushPlugin] didRegisterForRemoteNotificationsWithDeviceToken");
 
-#if !TARGET_IPHONE_SIMULATOR
+    if([self usesFCM]) {
+        NSLog(@"[PushPlugin] Setting APNS Token to FCM");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if([FIRApp defaultApp] == nil)
+                [FIRApp configure];
```

Comment:
I think this should be moved to `pluginInitialize` method, apps are supposed to call it on `didFinishLaunchingWithOptions`, we can't make the plugin call the method on `didFinishLaunchingWithOptions`, but `pluginInitialize` is the sooner we can. 

Also, I think that if we move it, we can also remove the `if([FIRApp defaultApp] == nil)` check since it was added to avoid multiple calls to `[FIRApp configure];` that could happen if calling `init` multiple times, but if we put it in `pluginInitialize`, then it won't be called multiple times.

Reply:
Commit: https://github.com/havesource/cordova-plugin-push/pull/285/commits/0e0928ef8a38fa812f383d94e77eed572483cdd7